sdk/communication/Azure.Communication.Chat/src/Shared/CommunicationIdentifierSerializer.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+
+namespace Azure.Communication
+{
+    internal class CommunicationIdentifierSerializer
+    {
+        private const string OrgIdPrefix = "8:orgid:";
+        private const string TeamsVisitorPrefix = "8:teamsvisitor:";
+
+        public static CommunicationIdentifier Deserialize(CommunicationIdentifierModel identifier)
+        {
+            var id = identifier.Id;
+            var kind = identifier.Kind;
+            return kind switch
+            {
+                IdentifierKinds.CommunicationUser
+                    => new CommunicationUserIdentifier(AssertNotNull(id, nameof(identifier.Id), kind)),
+                IdentifierKinds.CallingApplication
+                    => new CallingApplicationIdentifier(AssertNotNull(id, nameof(identifier.Id), kind)),
+                IdentifierKinds.PhoneNumber
+                    => new PhoneNumberIdentifier(AssertNotNull(identifier.PhoneNumber, nameof(identifier.PhoneNumber), kind)),
+                IdentifierKinds.MicrosoftTeamsUser
+                    => new MicrosoftTeamsUserIdentifier(
+                        GetTeamsUserId(AssertNotNull(id, nameof(identifier.Id), kind)),
+                        AssertNotNull(identifier.IsAnonymous, nameof(identifier.IsAnonymous), kind)),
+                IdentifierKinds.Unknown
+                    => new UnknownIdentifier(AssertNotNull(id, nameof(identifier.Id), kind)),
+                _
+                    => new UnknownIdentifier(AssertNotNull(id, nameof(identifier.Id), kind)),
```

Comment:
This is an interesting case because id is no longer guaranteed to be the full Mri.
For example if we didn't have MicrosoftTeamsUser in the client here yet and it gets added and sent from the service, the client wouldn't be able to roundtrip it because the id doesn't include the mri prefix.

For that we would have to preserve all other properties in the json object, or we live with the limitation that UnkownIdentifier is only a best-effort fallback but no guarantee.

(A badly over-engineered solution would be the service querying the user agent to determine whether to send down the new identifier or an UnknownIdentifier)

Reply:
We talked offline and decided to have `id` always be the full Mri